### PR TITLE
docs: add scn-tex documentation, add README

### PR DESCRIPTION
**Pull request by MksmOrlov**

### CODE REVIEW

File: docs/SCL_machine.tex [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+                        а каждое следующее множество правил применяется после предыдущего. Таким образом указываются приоритеты множеств правил.}
+                    \scnrelfrom{описание примера}{\scnfileimage{images/rules_set.png}}
+                \end{scnindent}
+                \scnitem{множество аргументов}
+                \begin{scnindent}
+                    \scnidtf{argument set}
+                    \scntext{пояснение}{Множество тех элементов, которые должны быть подставлены как значение переменных шаблона цели.}
+                    \begin{scnindent}
+                        \scntext{пояснение}{В данном примере значением переменной \_x в шаблоне цели может быть sc-узел argument\_1, тогда значением \_y будет argument\_2
+                        или наоборот: значением \_x sc-eузел argument\_2, а значением \_y -- argument\_1.}
```
Comment (AlexandrZagorskiy) [inline docs/SCL_machine.tex:47]:
```suggestion
                        \scntext{пояснение}{В данном примере значением переменной \_x в шаблоне цели может быть sc-узел argument\_1, тогда значением \_y будет argument\_2.
                        Также значением переменной \_x может быть sc-узел argument\_2, а значением \_y -- argument\_1.}
```

Comment (AlexandrZagorskiy) [reply docs/SCL_machine.tex:47]: What is the basis of the choice? How developer should understand and control it?

File: docs/SCL_machine.tex [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+                \end{scnindent}
+            \end{scnrelfromvector}
+            \scnrelfrom{пример выходной конструкции}{\scnfileimage{images/DirectInferenceAgent_output.png}}
+            \begin{scnrelfromvector}{обобщённый алгоритм}
+                \scnfileitem{Получение параметров агента, проверка их валидности. Вызов агента;}
+                \scnfileitem{Проверка, достигнута ли уже цель в базе знаний;}
+                \begin{scnindent}
+                    \scntext{примечание}{Выполняется поиск по шаблону target template с параметрами шаблона arguments set.}
+                \end{scnindent}
+                \scnfileitem{Построение вектора очереди правил на основе множества правил. Цикл по всем правилам и по не достигнута цель;}
```
Comment (AlexandrZagorskiy) [inline docs/SCL_machine.tex:59]:
```suggestion
                \scnfileitem{Построение вектора очереди правил на основе множества правил. Цикл по всем правилам пока не достигнута цель;}
```

**Review by AlexandrZagorskiy — CHANGES_REQUESTED** [review]

File: docs/SCL_machine.tex (hunk 1 above)
Comment (MksmOrlov) [reply docs/SCL_machine.tex:47]: Random. Without control

**Review by AlexandrZagorskiy — APPROVED** [review]

File: docs/scl-machine.tex [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+            \end{scnrelfromset}
+        \end{scnindent}
+        \scnfileitem{Возврат дерева применённых правил.}
+    \end{scnrelfromvector}
+\end{scnindent}
+
+\begin{scnrelfromset}{недостатки текущего состояния}
+    \scnfileitem{Импликация в текущем состоянии интерпретируется не как логическая связка, а как отношение \scnkeyword{выводимости*},
+    то есть импликация не возвращает логическую константу, а генерирует новые знания. Вместо этого должны использоваться правила вывода,
+    например, Modes ponens и другие правила, в процессе логического вывода.}
```
Comment (SHtress) [inline docs/scl-machine.tex:94]:
```suggestion
    например, Modus ponens и другие правила, в процессе логического вывода.}
```

**Review by SHtress — APPROVED** [review]

**Pull request merged**